src/Agent.Worker/JobRunner.cs

```diff
@@ -347,6 +348,9 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                 finally
                 {
                     Trace.Info("Finalize job.");
+                    
+                    PublishSecretsMaskingTelemetry(jobContext);
```

Comment:
>PublishSecretsMaskingTelemetry [](http://example.com/codeflow?start=20&length=30)

Is this a good place to put the secret masking telemetry? Is this when the effective lifetime of the masker is over? What are the lifetime semantics of the host context relative to this code?

Reply:
Looks good to me, considering we upload logs just after (UploadDiagnosticLogsAsync)